third_party/move/move-vm/runtime/src/interpreter.rs

```diff
@@ -2311,6 +2472,63 @@ impl Frame {
                         let field_ref = reference.borrow_field(offset)?;
                         interpreter.operand_stack.push(field_ref)?;
                     },
+                    Bytecode::ImmBorrowVariantField(idx) | Bytecode::MutBorrowVariantField(idx) => {
+                        let instr = match instruction {
+                            Bytecode::MutBorrowVariantField(_) => S::MutBorrowVariantField,
+                            _ => S::ImmBorrowVariantField,
+                        };
+                        gas_meter.charge_simple_instr(instr)?;
+
+                        let field_info = resolver.variant_field_info_at(*idx);
+                        let reference = interpreter.operand_stack.pop_as::<StructRef>()?;
+                        let field_ref = reference.borrow_variant_field(
+                            &field_info.variants,
+                            field_info.offset,
+                            &|v| {
+                                field_info
+                                    .definition_struct_type
+                                    .variant_name_for_message(v)
+                            },
+                        )?;
+                        interpreter.operand_stack.push(field_ref)?;
+                    },
+                    Bytecode::ImmBorrowVariantFieldGeneric(fi_idx)
+                    | Bytecode::MutBorrowVariantFieldGeneric(fi_idx) => {
+                        // TODO: Even though the types are not needed for execution, we still
```

Comment:
I think we should really add code owners, project names, or issues to todos. @vgao1996 this probably should be TODO(Gas)? Also, why this is a todo at all? Do we want to fix it? The comment says it is wasteful, but that's about it...